openstates/az/bills.py

```diff
@@ -213,6 +215,51 @@ def actor_from_action(self, bill, action, self_chamber):
                 else:
                     return action_map[key]
 
+    def scrape_votes(self, bill, page):
+        base_url = 'https://apps.azleg.gov/api/BillStatusFloorAction'
+        for header in page['FloorHeaders']:
+            params = {
+                'billStatusId': page['BillId'],
+                'billStatusActionId': header['BillStatusActionId'],
+                'includeVotes': 'true',
+            }
+            resp = self.get('{}?{}'.format(base_url, urllib.parse.urlencode(params)))
+            actions = json.loads(resp.content.decode('utf-8'))
+            for action in actions:
+                if action['Action'] == 'No Action':
+                    continue
+                action_date = datetime.datetime.strptime(action['ReportDate'], '%Y-%m-%dT%H:%M:%S')
+                vote = VoteEvent(
+                    chamber={
+                        'S': 'upper',
+                        'H': 'lower',
+                    }[header['LegislativeBody']],
+                    motion_text=action['Action'],
+                    classification='passage',
+                    result=(
+                        'pass'
+                        if action['UnanimouslyAdopted'] or action['Ayes'] > action['Nays']
+                        else 'fail'
+                    ),
+                    start_date=action_date.strftime('%Y-%m-%d'),
+                    bill=bill,
+                )
+                vote.add_source(resp.url)
+                vote.set_count('yes', action['Ayes'] or 0)
+                vote.set_count('no', action['Nays'] or 0)
+                vote.set_count('other', (
+                    (action['Present'] or 0) + (action['Absent'] or 0) +
+                    (action['Excused'] or 0) + (action['NotVoting'] or 0)))
+
+                for v in action['Votes']:
+                    vote_type = {
+                        'Y': 'yes',
+                        'N': 'no',
+                    }.get(v['Vote'], 'other')
```

Comment:
We generally prefer more brittle scraping patterns. This code could continue to "work" even if yes votes are changed from "Y" to "y" or "yes"